compiler/src/iree/compiler/API/Internal/IREEGPUDialectCAPI.cpp

```diff
@@ -350,3 +350,42 @@ MlirAttribute ireeGPULoweringConfigAttrGetMmaKind(MlirAttribute attr) {
 
   return wrap(mma_attr);
 }
+
+ireeGPUMMASingleSubgroupLayout
+ireeGPUGetSingleSubgroupLayout(MlirAttribute attr, uint32_t fragment) {
+  assert(ireeAttributeIsAGPUMMAIntrinsicAttr(attr) ||
+         ireeAttributeIsAGPUVirtualMMAIntrinsicAttr(attr) &&
+             "Expected MMA or VirtualMMA Intrinsic");
+
+  mlir::Attribute baseAttr = unwrap(attr);
+  mlir::iree_compiler::IREE::GPU::MMASingleSubgroupLayout layout;
+  mlir::iree_compiler::IREE::GPU::MMAFragment frag =
+      static_cast<mlir::iree_compiler::IREE::GPU::MMAFragment>(fragment);
+
+  if (auto intrinsicAttr =
+          llvm::dyn_cast<mlir::iree_compiler::IREE::GPU::MMAIntrinsicAttr>(
+              baseAttr)) {
+    layout = mlir::iree_compiler::IREE::GPU::getSingleSubgroupLayout(
+        intrinsicAttr.getValue(), frag);
+  } else if (auto virtualIntrinsicAttr = llvm::dyn_cast<
+                 mlir::iree_compiler::IREE::GPU::VirtualMMAIntrinsicAttr>(
+                 baseAttr)) {
+    layout = mlir::iree_compiler::IREE::GPU::getSingleSubgroupLayout(
+        virtualIntrinsicAttr.getValue(), frag);
+  } else {
+    assert(false &&
+           "Unreachable: attribute must be MMA or VirtualMMA intrinsic");
+  }
+
+  mlir::MLIRContext *context = baseAttr.getContext();
+  mlir::Builder builder(context);
+
+  ireeGPUMMASingleSubgroupLayout result;
```

Comment:
We can start by zero-initializing the whole struct in case we forgot to initialize some field later on. Zero values are usually easier to track down than uninitialized garbage.
```suggestion
  ireeGPUMMASingleSubgroupLayout result = {};
```